language/move-vm/runtime/src/native_functions.rs

```diff
@@ -167,6 +170,18 @@ impl<'a, 'b> NativeContext<'a, 'b> {
         self.extensions
     }
 
+    pub fn load_type(&self, type_tag: &TypeTag) -> VMResult<Type> {
```

Comment:
Interesting. What case do you not have a `Type`?

Reply:
Indeed, such an odd scenario would be the following:

Due to a circular dependency issue, I cannot `use` a module that defines the type of a resource I am interesting in reading. To deal with this, I would construct a `TypeTag` for that resource's type in the Rust native and use `load_type` on it. Then, I can read the resource using `load_resource`.

Reply:
Do you have a code example? I'm not quite sure how you are loading a value without having to refer to it's type at some point. 

(Note: It's a native function for whatever adapter you are working on, so some core assumptions in Move could go out the window, but it is worth mentioning there is an assumption that you only load resources in the module that defines them)

Reply:
Maybe this might not be such a good idea, then?

I don't have a code example but, as mentioned above, in my use case, I can easily construct a `StructTag` and hence a `TypeTag` for say a resource at `0x01::chain_id::ChainId`. I cannot get a `Type` though because I cannot `use` the `chain_id` module (due to a circular dependency issue), fetch the `ChainId` type and pass it down to a Rust native.